alma/views/js/alma-inpage.js

```diff
@@ -22,215 +22,166 @@
-            $('<div class="alert alert-danger">Error : .alma-inpage class not found in DOM</div>').insertBefore('#HOOK_PAYMENT');
-            console.log('No alma-inpage class found in DOM');
-            return
-        }
-    }
+    const eventAlma = async function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        await createPayment(url, inPage, input);
+    };
 
-    const almaInPageOnload = function () {
-        checkClassInDomInPage();
+    paymentButtonEvents.push(eventAlma);
+    paymentButton.addEventListener('click', eventAlma);
+}
 
-        //Prestashop 1.7+
-        $("input.ps-shown-by-js[name=payment-option]").click(function () {
-            removeCheckoutEvents();
-            if (inPage !== undefined) {
-                inPage.unmount();
-            }
-            if ($(this).data('module-name') === 'alma') {
-                let paymentOptionId = $(this).attr('id');
-                let selectorSetting = $('#' + paymentOptionId + '-additional-information .alma-inpage');
-                let showPayButton = false;
-                let installmentButton = selectorSetting.data('installment');
-                if (installmentButton === 1) {
-                    selectorSetting.hide();
-                }
-                if (selectorSetting.length > 0) {
-                    inPage = createIframe(paymentOptionId, selectorSetting, showPayButton);
-                    processAlmaPayment(paymentOptionId, inPage);
-                }
-            }
-        });
+async function createPayment(url, inPage, input = null) {
+    if (isAlmaPayment(url)) {
+        displayLoader();
+        try {
+            let response = await fetch(url);
+            let paymentData = await response.json();
 
-        //Prestashop 1.6-
-        $(".alma-inpage.ps16").click(function (e) {
-            e.preventDefault();
-            let paymentOptionId = $(this).attr('id');
-            let selectorSetting = $('#alma-inpage-' + paymentOptionId);
-            let showPayButton = true;
-            let url = $('#' + paymentOptionId).attr('href');
+            inPage.startPayment(
+                {
+                    paymentId: paymentData.id,
+                    onUserCloseModal: () => {
+                        let checkboxTermsOfService = document.querySelector('.ps-shown-by-js[type=checkbox]');
+                        if (checkboxTermsOfService !== null) {
+                            checkboxTermsOfService.checked = false;
+                        }
+                        document.querySelector('.alma-loader--wrapper').remove();
+                        if (input) {
+                            input.checked = false
+                            inPage.unmount();
+                        }
+                        removeAlmaEventsFromPaymentButton();
+                        onloadAlma();
+                    }
+                }
+            );
+        } catch(e) {
+            console.log(e);
```

Comment:
For later: could we use Sentry to log exceptions?

Reply:
How can I use Sentry for this log ?

Reply:
> For later: could we use Sentry to log exceptions?

For now we do not do that but it's a feature to study